azurerm/internal/services/compute/windows_virtual_machine_resource_data_disk_test.go

```diff
@@ -0,0 +1,770 @@
+}
+
+
+resource "azurerm_windows_virtual_machine" "test" {
+  name                = local.vm_name
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  size                = "Standard_F2"
+  admin_username      = "adminuser"
+  admin_password      = "P@$$w0rd1234!"
+
+  network_interface_ids = [
+    azurerm_network_interface.test.id,
+  ]
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  data_disks {
+    local {
+      name                   = "acctest-localdisk"
+      lun                    = 1
+      caching                = "ReadOnly"
+      storage_account_type   = "Standard_LRS"
+      disk_encryption_set_id = azurerm_disk_encryption_set.update.id
+      disk_size_gb           = 1
+    }
+  }
+
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2016-Datacenter"
+    version   = "latest"
+  }
+}
+
+`, template, data.RandomInteger)
+}
+
+func (r WindowsVirtualMachineResource) dataDisksMultipleUpdateSize(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_windows_virtual_machine" "test" {
+  name                = local.vm_name
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  size                = "Standard_F2"
+  admin_username      = "adminuser"
+  admin_password      = "P@$$w0rd1234!"
+
+  network_interface_ids = [
+    azurerm_network_interface.test.id,
+  ]
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  data_disks {
+    local {
+      name                 = "acctest-localdisk"
+      lun                  = 1
+      caching              = "ReadOnly"
+      storage_account_type = "Standard_LRS"
+      disk_size_gb         = 2
+    }
+
+    local {
+      name                 = "acctest-localdisk2"
+      lun                  = 2
+      caching              = "ReadOnly"
+      storage_account_type = "Standard_LRS"
+      disk_size_gb         = 1
+    }
+  }
+
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2016-Datacenter"
+    version   = "latest"
+  }
+}
+
+`, template)
+}
+
+func (r WindowsVirtualMachineResource) dataDisksExistingDisk(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_managed_disk" "test" {
+  name                 = "acctested-%d"
+  location             = azurerm_resource_group.test.location
+  resource_group_name  = azurerm_resource_group.test.name
+  storage_account_type = "Standard_LRS"
+  create_option        = "Empty"
+  disk_size_gb         = "1"
```

Comment:
we should add a test (to both Linux and Windows) which increases the size of the external disk too - whilst there's a test for that in the Managed Disk resource tests, perhaps it's worth updating an internal and external at the same time to validate the locking works (since that stops/starts the VM)?